tools/reviewer/job/sync/GithubWriter.java

```diff
@@ -16,201 +16,116 @@
 
 package com.google.startupos.tools.reviewer.job.sync;
 
-import com.google.startupos.tools.reviewer.job.sync.GithubProtos.CreatePullRequestCommentRequest;
-import com.google.startupos.tools.reviewer.job.sync.GithubProtos.CreatePullRequestCommentRequestData;
+import com.google.startupos.tools.reviewer.job.sync.GithubProtos.CreatePullRequestReviewRequest;
+import com.google.startupos.tools.reviewer.job.sync.GithubProtos.CreatePullRequestReviewRequestData;
 import com.google.startupos.tools.reviewer.job.sync.GithubProtos.CreatePullRequestRequest;
 import com.google.startupos.tools.reviewer.job.sync.GithubProtos.CreatePullRequestRequestData;
-import com.google.startupos.tools.reviewer.job.sync.GithubProtos.CreatePullRequestResponse;
-import com.google.startupos.tools.reviewer.job.sync.GithubProtos.CreateReviewCommentRequest;
-import com.google.startupos.tools.reviewer.job.sync.GithubProtos.CreateReviewCommentRequestData;
-import com.google.startupos.tools.reviewer.job.sync.GithubProtos.PullRequestRequest;
-import com.google.startupos.tools.reviewer.job.sync.GithubProtos.File;
-import com.google.startupos.tools.reviewer.job.sync.GithubProtos.PullRequest;
-import com.google.startupos.tools.reviewer.job.sync.GithubProtos.PullRequestFilesRequest;
-import com.google.startupos.tools.reviewer.job.sync.GithubProtos.PullRequestsRequest;
-import com.google.startupos.tools.reviewer.localserver.service.Protos.Comment;
-import com.google.startupos.tools.reviewer.localserver.service.Protos.Diff;
-import com.google.startupos.tools.reviewer.localserver.service.Protos.Thread;
+import com.google.startupos.tools.reviewer.job.sync.GithubPullRequestProtos.PullRequest;
+import com.google.startupos.tools.reviewer.job.sync.GithubPullRequestProtos.Review;
+import com.google.startupos.tools.reviewer.job.sync.GithubPullRequestProtos.IssueComment;
 
-import java.io.IOException;
-import java.time.Instant;
-import java.util.List;
-
-/** Writes `tools.reviewer.localserver.service.Protos.Diff` to GitHub using GitHub API */
+/**
+ * Writes `tools.reviewer.job.sync.GithubPullRequestProtos.PullRequest` to GitHub using GitHub API
+ */
 public class GithubWriter {
-  private final String reviewerDiffLink;
   private final GithubClient githubClient;
 
-  GithubWriter(GithubClient githubClient, String reviewerDiffLink) {
+  GithubWriter(GithubClient githubClient) {
     this.githubClient = githubClient;
-    this.reviewerDiffLink = reviewerDiffLink;
   }
 
-  public void writeDiff(Diff diff, String repoOwner, String repoName) throws IOException {
-    int diffNumber = getGithubPullRequestNumber(diff, repoOwner, repoName);
+  public void writePullRequest(PullRequest pullRequest, String repoOwner, String repoName) {
```

Comment:
The method here should also accept the old `PullRequest` (presumably the one currently on GitHub), and only update what's changed.
So you would check the difference in the fields, and based on what changed, update GitHub. This is to avoid sending many unneeded requests when nothing has changed.